submission/distributor_test.go

```diff
@@ -354,23 +382,23 @@ func TestDistributorAddPreChain(t *testing.T) {
 			ctx, cancel := context.WithTimeout(context.Background(), time.Second)
 			defer cancel()
 
-			dist.Run(ctx)
+			if errs := dist.RefreshRoots(ctx); len(errs) != 1 || errs["ct.googleapis.com/icarus/"] == nil {
+				// 1 error is expected, because the Icarus log has an invalid root (see RootCerts).
+				t.Fatalf("dist.RefreshRoots() = %v, want 1 error for 'ct.googleapis.com/icarus/'", errs)
+			}
 
 			scts, err := dist.AddPreChain(context.Background(), tc.rawChain)
 			if gotErr := (err != nil); gotErr != tc.wantErr {
-				t.Errorf("Expected to get errors is %v while actually getting errors is %v", tc.wantErr, gotErr)
+				t.Fatalf("dist.AddPreChain(%q) = (_, %v), want err? %t", tc.rawChain, err, tc.wantErr)
```

Comment:
Are you sure you want to include bulky []byte rawChain into error message?

Reply:
It does result in a rather large error message, but it might be helpful - it would, for example, make it obvious if the chain was not being passed to AddPreChain() correctly.